### DISCUSSION TITLE
Continue function behavior

### DISCUSSION THREAD

**Discussion by Tedy50** [oobabooga/text-generation-webui#4948] [Ideas]
Sometimes AI response does not fit into token limit  and there is "continue" function to overcome that problem but to work properly this function should use the same seed as before or it may mess up the response.

so the request is to make it use the same seed as before instead of generating something new

I think it may be even good idea to retain the same seed over entire conversation session until one hits "regenerate" as there is no benefit from changing it and as I noticed it is messing up AI "plans"

**Comment by mabrown666** [comment]
The LLM uses the prompt plus the generated text so far to produce the next word. It doesn't have a "plan". Without including the generated text it cannot continue.
You can see this if you use the notepad interface, and the precise preset from parameters tab. Give it a prompt, let it generate. Then delete the last half of the generated text (not the prompt) and it will regenerate the same text. But if you repeat this but add a different word to the generated text (especially words like however or but) it will generate a different response.

It is the prompt and the resulting output that must fit into the context, not just the prompt.

**Reply by Tedy50** [reply]
It is not entirely correct, because LLM indeed does have a plan which is actually preprocessed context 
even if it is making one word at a time those words follow the plan.
like if you tell AI to generate 10 names it will  keep in mind that it has to be 10 names and generate names until there are 10 of them.
If you place a quote mark it knows that it has to produce some text and end with another quote mark.

If you give AI a prompt to generate text it will do it pretty well but if you decide to edit the text and ask AI continue, it will get pretty confused and start generating nonsense because the edited text no longer fits into its plan.

I noticed that if you terminate generation and let it continue with different seen it will often start generating entirely different story
